zerver/tests/test_subs.py

```diff
@@ -591,6 +591,45 @@ def test_set_color_missing_color(self):
         self.assert_json_error(
             result, "value key is missing from subscription_data[0]")
 
+    def test_set_pin_to_top(self):
+        # type: () -> None
+        """
+        A POST request to /json/subscriptions/property with stream_name and
+        pin_to_top data pins the stream.
+        """
+        test_email = "hamlet@zulip.com"
+        self.login(test_email)
+
+        old_subs, _ = gather_subscriptions(get_user_profile_by_email(test_email))
+        sub = old_subs[0]
+        stream_name = sub['name']
+        new_pin_to_top = not sub['pin_to_top']
+        result = self.client.post(
+            "/json/subscriptions/property",
+            {"subscription_data": ujson.dumps([{"property": "pin_to_top",
+                                                "stream": stream_name,
+                                                "value": new_pin_to_top}])})
+
+        self.assert_json_success(result)
+
+        new_subs = gather_subscriptions(get_user_profile_by_email(test_email))[0]
+        found_sub = None
+        for sub in new_subs:
+            if sub['name'] == stream_name:
+                found_sub = sub
+                break
+
+        self.assertIsNotNone(found_sub)
+        self.assertEqual(found_sub['pin_to_top'], new_pin_to_top)
+
+        new_subs.remove(found_sub)
+        for sub in old_subs:
+            if sub['name'] == stream_name:
+                found_sub = sub
+                break
+        old_subs.remove(found_sub)
+        self.assertEqual(old_subs, new_subs)
```

Comment:
I think this last section of the test (checking the existing subs weren't changed) is unnecessary.  We sorta already check that with the events test.